cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -179,30 +182,155 @@ def __init__(self, scope: core.Construct, id: str, head_node: HeadNode, **kwargs
+
+        if not fsx_id and shared_fsx.mount_dir.value:
+            fsx_resource = fsx.CfnFileSystem(
+                scope=self,
+                storage_capacity=shared_fsx.storage_capacity.value,
+                lustre_configuration=fsx.CfnFileSystem.LustreConfigurationProperty(
+                    deployment_type=shared_fsx.deployment_type.value,
+                    imported_file_chunk_size=shared_fsx.imported_file_chunk_size.value,
+                    export_path=shared_fsx.export_path.value,
+                    import_path=shared_fsx.import_path.value,
+                    weekly_maintenance_start_time=shared_fsx.weekly_maintenance_start_time.value,
+                    automatic_backup_retention_days=shared_fsx.automatic_backup_retention_days.value,
+                    copy_tags_to_backups=shared_fsx.copy_tags_to_backups.value,
+                    daily_automatic_backup_start_time=shared_fsx.daily_automatic_backup_start_time.value,
+                    per_unit_storage_throughput=shared_fsx.per_unit_storage_throughput.value,
+                    auto_import_policy=shared_fsx.auto_import_policy.value,
+                    drive_cache_type=shared_fsx.drive_cache_type.value,
+                ),
+                backup_id=shared_fsx.backup_id.value,
+                kms_key_id=shared_fsx.kms_key_id.value,
+                id=id,
+                file_system_type="LUSTRE",
+                storage_type=shared_fsx.drive_cache_type.value,
+                subnet_ids=self._cluster.compute_subnet_ids,
+                security_group_ids=self._cluster.compute_security_groups,
+            )
+            fsx_id = fsx_resource.ref
+
+        return fsx_id
+
+    def _add_efs_storage(self, id: str, shared_efs: SharedEfs):
+        """Add specific Cfn Resources to map the EFS storage."""
+        # EFS FileSystem
+        efs_id = shared_efs.file_system_id.value
+        if not efs_id and shared_efs.mount_dir:
+            efs_resource = efs.CfnFileSystem(
+                scope=self,
+                id=id,
+                encrypted=shared_efs.encrypted.value,
+                kms_key_id=shared_efs.kms_key_id.value,
+                performance_mode=shared_efs.performance_mode.value,
+                provisioned_throughput_in_mibps=shared_efs.provisioned_throughput.value,
+                throughput_mode=shared_efs.throughput_mode.value,
+            )
+            efs_id = efs_resource.ref
+
+        checked_availability_zones = []
+        # Mount Targets for Compute Fleet
+        compute_subnet_ids = self._cluster.compute_subnet_ids
+        for subnet_id in compute_subnet_ids:
+            self._add_efs_mount_target(
+                id, efs_id, self._cluster.compute_security_groups, subnet_id, checked_availability_zones
+            )
+
+        # Mount Target for Head Node
+        self._add_efs_mount_target(
+            id,
+            efs_id,
+            self._cluster.head_node.networking.security_groups.value,
```

Comment:
security_groups parameter corresponds to the old vpc_security_group parameter that has a value only if the user is setting it.
By default security_groups is None so you should take it from the one created by us in the main template.

Reply:
Fixed by adding a collection `_compute_security_groups` to ClusterStack. In this collection we will add both security groups created in the main substack and the ones explicitly set by users. All storage resources will refer to this list.